database/src/main/java/com/firebase/ui/database/FirebaseIndexArray.java

```diff
@@ -67,15 +67,15 @@ private int getIndexForKey(String key) {
             String superKey = super.getItem(keyIndex).getKey();
             if (key.equals(superKey)) {
                 break;
-            } else if (getItem(index).getKey().equals(superKey)) {
+            } else if (mDataSnapshots.get(index).getKey().equals(superKey)) {
                 index++;
             }
         }
         return index;
     }
 
     private boolean isMatch(int index, String key) {
-        return index >= 0 && index < getCount() && getItem(index).getKey().equals(key);
+        return index >= 0 && index < getCount() && mDataSnapshots.get(index).getKey().equals(key);
```

Comment:
We should have never used `getItem()` here because it's public api and anyone overriding that method will break our implementation.